src/scanner.c

```diff
@@ -1,10 +1,25 @@
+#include <string.h>  // memcpy()
+#include <wctype.h>  // iswspace()
+
+#include "tree_sitter/parser.h"
+
+// Uncomment if debugging for extra output during parsing. Note that we can't
+// use `vprintf()` for print debugging in WASM or on CRAN for the R package!
+// #define TREE_SITTER_R_DEBUG
 
-#include <ctype.h>   // isspace()
+#ifdef TREE_SITTER_R_DEBUG
 #include <stdarg.h>  // va_list, va_start(), va_end()
-#include <stdio.h>   // printf()
-#include <stdlib.h>  // getenv()
-#include <string.h>  // memcpy()
-#include <tree_sitter/parser.h>
+#include <stdio.h>   // vprintf()
+
+static inline void debug_print(const char* fmt, ...) {
```

Comment:
Could we make use of `Rprintf()` for R packages when debugging? Something like:

```
#if defined(R_VERSION)
# define debug_print Rprintf
#elif defined(__wasm__)
# define debug_print
#else
// regular code
#endif
```

In other words, if we could toggle debug output at runtime via a flag or something similar, I think that would be ideal.

Reply:
To me these debug things are only useful with `tree-sitter parse -d` where it gets intermingled with the other debug logging that tree sitter emits, otherwise you don't really have any line/col info about where its coming from, like look for `something is wrong!` here, which i emit with `debug_print()`

```
davis@davisvaughan-01HQ tree-sitter-r % tree-sitter parse ~/Desktop/test2.R -d
new_parse
process version:0, version_count:1, state:1, row:0, col:0
lex_external state:2, row:0, column:0
lex_internal state:0, row:0, column:0
  consume character:'t'
  consume character:'e'
  consume character:'s'
  consume character:'t'
lexed_lookahead sym:_identifier, size:4
shift state:2560
process version:0, version_count:1, state:2560, row:0, col:4
lex_external state:28, row:0, column:4
  consume character:'('
lexed_lookahead sym:(, size:1
reduce sym:identifier, child_count:1
reduce sym:_expression, child_count:1
shift state:1739
process version:0, version_count:1, state:1739, row:0, col:5
lex_external state:31, row:0, column:5
something is wrong!
  consume character:')'
lexed_lookahead sym:), size:1
shift state:2866
process version:0, version_count:1, state:2866, row:0, col:6
```

I'm not sure having an `Rprintf()` call would get us much since we don't see the other debug output

Reply:
Gotcha -- in that case, I think it's not worth the effort to make those changes.